### PR TITLE
feat(api): add evaluation endpoint for conversations

### PR DESCRIPTION
**Pull request by geclos**
- Introduced a new POST /evaluate endpoint to handle evaluation requests for conversations.
- Implemented the evaluateHandler to process evaluation requests, including optional evaluation UUIDs.
- Added unit tests to cover various scenarios such as unauthorized access, evaluating all evaluations, evaluating specific evaluations, handling no evaluations, and invalid conversation UUIDs.
- Updated the SDK to include a new eval method for evaluating conversations.
- Modified job definitions to handle optional batchId in runEvaluationJob.
- Enhanced the evaluateDocumentLog service to enqueue evaluation jobs.

This change allows users to evaluate conversations through a dedicated endpoint, providing flexibility to evaluate all or specific evaluations. It also ensures proper handling of different scenarios and integrates the functionality into the SDK for easier usage.

### CODE REVIEW

File: packages/sdks/typescript/src/utils/index.ts [hunk 1]
```diff
@@ -58,6 +62,7 @@ export class RouteResolver {
     const base = `${this.baseUrl}/conversations`
     return {
       chat: (uuid: string) => `${base}/${uuid}/chat`,
+      evaluate: (uuid: string) => `${base}/${uuid}/evaluate`,
```
Comment (andresgutgon) [inline packages/sdks/typescript/src/utils/index.ts:65]: Missing tests

File: packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.ts [hunk 2]
```diff
@@ -55,7 +55,10 @@ export async function runEvaluationJob(job: Job<RunEvaluationJobData>) {
   const { workspaceId, batchId, documentUuid, documentLogUuid, evaluationId } =
     job.data
   const websockets = await WebsocketClient.getSocket()
-  const progressTracker = new ProgressTracker(await queues(), batchId)
+  let progressTracker: ProgressTracker | undefined
+  if (batchId) {
```
[the remark this answers is not in the document]
Comment (geclos) [reply packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.ts:59]: when called form the evaluate handler

File: packages/sdks/typescript/src/utils/index.ts (hunk 1 above)
Comment (geclos) [reply packages/sdks/typescript/src/utils/index.ts:65]: 💯

File: packages/sdks/typescript/src/utils/syncRun.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export async function syncRun(
     })
 
     onError?.(error)
-    return !onError ? Promise.reject(error) : Promise.resolve()
+    return !onError ? Promise.reject(error) : Promise.resolve(undefined)
```
Comment (geclos) [inline packages/sdks/typescript/src/utils/syncRun.ts:67]: bug, without this undefined we were returning type void type which ain't correct

File: docs/guides/sdk/javascript-typescript-sdk.mdx [hunk 4]
```diff
@@ -211,3 +211,45 @@ if (last.role === MessageRole.assistant && last.toolCalls.length > 0) {
...
+        type: 'text',
+        text: 'Tell me another joke about doctors',
+      },
+    ],
+  },
+])
+
+// 3. Evaluate the full conversation
+await sdk.eval(uuid)
+```
```
Comment (csansoon) [inline docs/guides/sdk/javascript-typescript-sdk.mdx:250]: Does this evaluate the log for ALL evaluations connected to the document?

Comment (geclos) [reply docs/guides/sdk/javascript-typescript-sdk.mdx:250]: I love the caps in "ALL" for an extra those of drama. Yes it does but you can pass it an optional array of evaluation uuids to filter by the evaluations that you are interested in if you only wanna run a subset

**Review by csansoon — APPROVED** [review]

File: docs/guides/sdk/javascript-typescript-sdk.mdx (hunk 4 above)
Comment (geclos) [reply docs/guides/sdk/javascript-typescript-sdk.mdx]: updated the docs

File: packages/sdks/typescript/src/utils/syncRun.ts (hunk 3 above)
Comment (andresgutgon) [reply packages/sdks/typescript/src/utils/syncRun.ts:67]: But was causing errors? Just curious

Comment (geclos) [reply packages/sdks/typescript/src/utils/syncRun.ts:67]: it causes typescript static errors

Comment (andresgutgon) [reply packages/sdks/typescript/src/utils/syncRun.ts:67]: 👍

**Pull request merged**